cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -729,6 +740,81 @@ class SimpleArray
     value_type * m_body = nullptr;
 }; /* end class SimpleArray */
 
+template <typename T>
+template <typename I>
+SimpleArray<T> SimpleArray<T>::take_along_axis(SimpleArray<I> const & indices)
+{
+    static_assert(std::is_integral_v<I>, "I must be integral type");
+    if (indices.ndim() != 1)
+    {
+        throw std::runtime_error(Formatter() << "SimpleArray: take_along_axis() supports only "
+                                                "in 1D array but the index array is "
+                                             << indices.ndim() << " dimension");
+    }
+    if (ndim() != 1)
+    {
+        throw std::runtime_error(Formatter() << "SimpleArray: take_along_axis() supports only "
+                                                "in 1D array but the array is "
+                                             << ndim() << " dimension");
+    }
+    if (shape()[0] != indices.shape()[0])
+    {
+        throw std::runtime_error(Formatter() << "SimpleArray: take_along_axis() supports same "
+                                                "shape of indices and array. Array size "
+                                             << shape()[0] << " != indices array size " << indices.shape()[0]);
+    }
+
+    SimpleArray<T> ret(*this);
+    for (size_t i = 0; i < shape()[0]; ++i)
+    {
+        ret.at(i) = at(static_cast<size_t>(indices[i]));
```

Comment:
The bound-checking is not necessary.  Please use a smarter way to implement the element accessing without the bound-checking.